### ISSUE TITLE
[apex] Provide access to underlying query of SoqlExpression

### ISSUE THREAD

**Issue by r-r-a-j** [pmd/pmd#2136]
![image](https://user-images.githubusercontent.com/33902071/69810301-7bc55600-11e3-11ea-84c1-c60fdb5391da.png)

**PMD Version:** 
6.19

**Description:** 
(New to PMD & Xpath so please forgive a newbie question!)

I can see the SoqlExpression in the AST but I need to parse the SOQL itself, however there is no Image attribute (is this the correct attribute I should be looking for??), nor is there any other further break down element that I can interrogate. I wish to avoid writing a Java class if at all possible and simply use Xpath.

**Code Sample demonstrating the issue:**
```java
public class Foo 
{
	public void test1() 
	{
	    Account acc = [Select 1 FROM Account ];
	    Account acc = [Select 2 FROM Account where x = x];
	    Account acc = [Select 2 FROM Account where x = x WITH SECURITY_ENFORCED];
	}
}
```
XPath-Expression: `//SoqlExpression`

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
Designer from CLI

**Comment by adangel** [comment]
Hi @r-r-a-j ,
that's not possible at the moment - but it's easy to add. The SoqlExpression node has access to the underlying query, but that is just not exposed as an xpath attribute.

Note, that the query is a simple string - there is no further structure. But you could then write a rule e.g. that finds all queries with this "security enforced" flag:

`//SoqlExpression[contains(@Query, 'WITH SECURITY_ENFORCED')]`

I've created a PR for that: #2137
